charts/aspnetcore/templates/deployment-hpa.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if ((.Values.autoscaling).enabled) }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ default (printf "%s-deployment" .Release.Name) .Values.autoscaling.name }}
+  labels:
+    {{- include "aspnetcore.standardLabels" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+      {{- toYaml .Values.commonLabels | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- with .Values.commonAnnotations }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  metrics:
+    - resource:
+        name: cpu
+        target:
+          averageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
+          type: Utilization
+      type: Resource
+  minReplicas: {{ max (add ((.Values.podDisruptionBudget).minAvailable) 1) (.Values.autoscaling.minReplicas) }}
+  maxReplicas: {{ max (add ((.Values.podDisruptionBudget).minAvailable) 1) (.Values.autoscaling.minReplicas) (.Values.autoscaling.maxReplicas) }}
```

Comment:
@Pycnomerus  Sorry if this comes out as rude but one of the reasons I dislike helm is that it encourages this kind of "template soup" where more and more complex logic starts piling up in templates. I also think most of this logic is kind of redundant.

I did some research and it seems like you can use JSON Schema to validate values files but the logic is not powerful enough to express the validation being done here. I think that maybe creating a `templates/prechecks.tpl` or something along those lines and running validations like this:

```
{{- if lt .Values.hpa.minReplicas .Values.pdb.minAvailable }}
{{- fail "hpa.minReplicas cannot be less than pdb.minAvailable" }}
{{- end }}
```

Would be more valuable as we inform the user of the problem and avoid complex logic trying to fix the user's mistake.

Reply:
@starcraft66 it's not rude that's a great ideal.

I have the same feeling about helm, for a lot of different reasons.

having a way of doing go validation is pretty neat tough, that make it so the template can remain more straightforward (as much as it can be with the manual indenting anyway

Reply:
Thanks, I'll try out your changes today.